qutip/core/dimensions.py

```diff
@@ -449,7 +455,7 @@ def __init__(self, dims):
         self._pure_dims = True
         self.__setitem__ = _frozen
 
-    def __eq__(self, other):
+    def __eq__(self, other) -> bool:
```

Comment:
Do we need to typehint `__eq__` and similar dunder methods that already have expected types in the language?

Reply:
That is true.